ArmaforcesMissionBot/Modules/Signups.cs

```diff
@@ -619,20 +619,27 @@ public async Task CancelMission(int missionNo)
         }
 
         [Command("edytuj-misje")]
-        [Summary("Po podaniu indeksu misji jako parametru włączy edycje danej misji (część bez zespołów).")]
+        [Summary("Po użyciu #wzmianki kanału misji jako parametru włączy edycje danej misji (część bez zespołów).")]
         [ContextDMOrChannel]
-        public async Task EditMission(int missionNo)
+        public async Task EditMission(IGuildChannel channel)
         {
             var currentlyEditedMission = SignupsData.GetCurrentlyEditedMission(Context.User.Id);
 
             if (currentlyEditedMission == null)
             {
-                var userMissions = SignupsData.Missions.Where(x => x.Owner == Context.User.Id);
-                var missionToBeEdited = userMissions.ElementAt(missionNo);
+                var missionToBeEdited = SignupsData.Missions.FirstOrDefault(x => x.SignupChannel == channel.Id);
+                if (missionToBeEdited == null)
+                {
+                    await ReplyAsync($"Nie ma misji o takiej nazwie.");
+                    return;
+                }    
+                
+                if (missionToBeEdited.Owner != Context.User.Id)
+                {
+                    await ReplyAsync($"Nie nauczyli żeby nie ruszać nie swojego?");
+                    return;
+                }
 
-                // Don't want to write another function just to copy class, and performance isn't a problem here so just serialize it and deserialize
-                var serialized = JsonConvert.SerializeObject(missionToBeEdited);
-                SignupsData.BeforeEditMissions[Context.User.Id] = JsonConvert.DeserializeObject<ArmaforcesMissionBotSharedClasses.Mission>(serialized);
```

Comment:
Why was this removed? Did you investigate it does nothing and can be safely removed or it was just an accident? If it doesn't break anything then that's great, two useless lines less.

I'm really curious if the `SignupsData.BeforeEditMissions` is used anywhere or is just some legacy code.

Reply:
My mistake as I over thought that. Code used for saving mission state before editing in case of cancelling. Done in this way to have pure data, not a reference.